.github/workflows/windows.yml

```diff
@@ -91,11 +91,20 @@ jobs:
 
 
       # Runs a set of commands using the runners shell
-      - name: Build
+      - name: Build system
```

Comment:
This is running each set of tests in sequence.  It should be possible to set up a matrix so they run as separate jobs, hopefully in parallel.

Reply:
Can be set up using multiple jobs.  See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions

Reply:
I agree. But as the step only takes c. 10s, is it really worth the bother?

Reply:
The issue is that any failure in this step will hide a pass or fail in the next step as each step will only run when all previous steps pass (unless one specifies fails are allowed, which hides failures).  Having the share and system calls as distinct jobs means they both run in all cases.

Reply:
I've split it up into a setup job and two build jobs, but the latter are failing, as they are missing all the stuff from the setup job. At the moment, I don't understand how this is going to work.

Reply:
I suspect they each need to be self-contained.  I can wrangle the config once this PR is merged (prob not today).  Thanks for the PR and all the updates.